assets/service_now_notifications_handler/index.py

```diff
@@ -197,16 +217,66 @@ def _get_parameter(self, parameter_name: str) -> Optional[str]:
         Returns:
             Parameter value or None if retrieval fails
         """
+        if not parameter_name:
+            logger.error("Parameter name is empty or None")
+            return None
+
         try:
+            logger.info(f"Retrieving parameter: {parameter_name}")
             response = self.ssm_client.get_parameter(
-                Name=parameter_name,
-                WithDecryption=True
+                Name=parameter_name, WithDecryption=True
             )
-            return response['Parameter']['Value']
+            return response["Parameter"]["Value"]
         except ClientError as e:
             error_code = e.response["Error"]["Code"]
+            error_message = e.response["Error"]["Message"]
+            if error_code == "AccessDeniedException":
+                logger.error(
+                    f"Access denied when retrieving parameter {parameter_name}. Check IAM permissions for this Lambda function."
+                )
+            elif error_code == "ParameterNotFound":
+                logger.error(
+                    f"Parameter {parameter_name} not found. Verify the parameter exists in SSM Parameter Store."
+                )
+            else:
+                logger.error(
+                    f"Error retrieving parameter {parameter_name}: {error_code} - {error_message}"
+                )
+            return None
+        except Exception as e:
             logger.error(
-                f"Error retrieving parameter {parameter_name}: {error_code}")
+                f"Unexpected error retrieving parameter {parameter_name}: {str(e)}"
+            )
+            return None
+
+
+class SecretsManagerService:
```

Comment:
nit(non-blocking): I’m not sure I see the value in wrapping SDK calls without additional logic. I think I just add an unnecessary layer of abstraction.